### PR TITLE
feat: remove D205 to enable PEP257 Docstring Conventions

### PR DESCRIPTION
**Pull request by ahsan-z-khan**
*Issue #, if available:* 1981

*Description of changes:* Removed D205 from pydocstyle to follow the PEP257 docstring conventions for future docstrings.

*Testing done:* Unit tests

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/session.py [hunk 1]
```diff
@@ -462,8 +463,6 @@ def train(  # noqa: C901
         debugger_hook_config=None,
         tensorboard_output_config=None,
         enable_sagemaker_metrics=None,
-        profiler_rule_configs=None,
```
Comment (metrizable) [inline src/sagemaker/session.py:465]: these cannot be removed

**Review by metrizable — CHANGES_REQUESTED** [review]

File: src/sagemaker/xgboost/estimator.py [hunk 2]
```diff
@@ -38,7 +38,8 @@ class XGBoost(Framework):
     It can also handle training using customer provided XGBoost entry point script.
     """
 
-    _framework_name = defaults.XGBOOST_NAME
+    # _framework_name = defaults.XGBOOST_NAME
+    _framework_name: str = defaults.XGBOOST_NAME
```
Comment (metrizable) [inline src/sagemaker/xgboost/estimator.py:42]: let's introduce type annotations outside of this PR

**Review by metrizable — CHANGES_REQUESTED** [review]

File: src/sagemaker/xgboost/estimator.py (hunk 2 above)
Comment (ahsan-z-khan) [reply src/sagemaker/xgboost/estimator.py:42]: will do! changed it to previous one

File: src/sagemaker/session.py [hunk 3]
```diff
@@ -195,7 +192,6 @@ def upload_data(self, path, bucket=None, key_prefix="data", extra_args=None):
                 Similar to ExtraArgs parameter in S3 upload_file function. Please refer to the
                 ExtraArgs parameter documentation here:
                 https://boto3.amazonaws.com/v1/documentation/api/latest/guide/s3-uploading-files.html#the-extraargs-parameter
-
```
Comment (metrizable) [inline src/sagemaker/session.py:198]: this line should not be removed

File: src/sagemaker/session.py [hunk 4]
```diff
@@ -278,9 +273,7 @@ def download_data(self, path, bucket, key_prefix="", extra_args=None):
                 download operation. Please refer to the ExtraArgs parameter in the boto3
                 documentation here:
                 https://boto3.amazonaws.com/v1/documentation/api/latest/guide/s3-example-download-file.html
-
```
Comment (metrizable) [inline src/sagemaker/session.py:281]: same with this and elsewhere. blank line before `Returns`

File: src/sagemaker/session.py [hunk 5]
```diff
@@ -280,7 +277,6 @@ def download_data(self, path, bucket, key_prefix="", extra_args=None):
                 https://boto3.amazonaws.com/v1/documentation/api/latest/guide/s3-example-download-file.html
 
         Returns:
-
```
Comment (metrizable) [inline src/sagemaker/session.py:283]: i think you should not have a `Returns:` section if you're not going to put anything in it, or have:

```
Returns:
    None
```

File: src/sagemaker/session.py [hunk 6]
```diff
@@ -1586,7 +1565,6 @@ def auto_ml(
                 pairs.
 
         Returns:
-
```
Comment (metrizable) [inline src/sagemaker/session.py:1589]: same comment as above

File: src/sagemaker/session.py (hunk 5 above)
Comment (ahsan-z-khan) [reply src/sagemaker/session.py:283]: ah that's understandable. I didn't remove it.

File: src/sagemaker/session.py [hunk 7]
```diff
@@ -2906,14 +2876,13 @@ def create_endpoint_config_from_existing(
...
-        """Create an Amazon SageMaker endpoint configuration from an existing one. Updating any
-        values that were passed in.
+        """Create an Amazon SageMaker endpoint configuration from an existing one.
 
+        IT also updates any values that were passed in.
         The endpoint configuration identifies the Amazon SageMaker model (created using the
         ``CreateModel`` API) and the hardware configuration on which to deploy the model. Provide
         this endpoint configuration to the ``CreateEndpoint`` API, which then launches the
         hardware and deploys the model.
-
```
Comment (metrizable) [inline src/sagemaker/session.py:2916]: need this blank line

File: src/sagemaker/session.py [hunk 8]
```diff
@@ -2906,14 +2876,13 @@ def create_endpoint_config_from_existing(
         new_data_capture_config_dict=None,
         new_production_variants=None,
     ):
-        """Create an Amazon SageMaker endpoint configuration from an existing one. Updating any
-        values that were passed in.
+        """Create an Amazon SageMaker endpoint configuration from an existing one.
 
+        IT also updates any values that were passed in.
```
Comment (metrizable) [inline src/sagemaker/session.py:2881]:
```suggestion
        It also updates any values that were passed in.
```

**Review by metrizable — CHANGES_REQUESTED** [review]

**Review by metrizable — APPROVED** [review]

**Pull request merged**